ui/frontend/BuildMenu.tsx

```diff
@@ -55,7 +57,11 @@ const BuildMenu: React.SFC<BuildMenuProps> = props => {
         Build and show the resulting LLVM IR, LLVM’s intermediate representation.
       </ButtonMenuItem>
       <ButtonMenuItem name="MIR" onClick={compileToMir}>
-        Build and show the resulting MIR, Rust’s intermediate representation.
+        Build and show the resulting MIR, Rust’s intermediate representation for control flow.
+      </ButtonMenuItem>
+      <ButtonMenuItem name="HIR" onClick={compileToHir}>
+        Build and show the resulting HIR, Rust’s intermediate representation for items in the current crate.
```

Comment:
"for items in the current crate" is kind of vague. I think something along the lines of

```suggestion
        Build and show the resulting HIR, Rust’s syntax-based intermediate representation.
```

would be clearer.

Reply:
I don't think this is very accurate; or at least, it doesn't make clear the difference between the AST and the HIR. I'm not 100% sure myself what the difference is.

Reply:
The HIR is a desugared version of the AST IIUC.

Reply:
If it's desugared, how can it also be syntax-based?

Reply:
It's a subset of Rust's syntax. There's no `for` loop, but to my knowledge it's a strict subset of the full Rust syntax.